src/odemis/gui/cont/multi_point_correlation.py

```
@@ -807,34 +808,63 @@ def _on_target_changes(self, targets: List[Target]) -> None:
-            self._tab_data_model.main.currentTarget.value.coordinates.value[2] = z_p
+            pixel_coords = get_pixel_3d_coordinates(self.correlation_target.fm_streams[0], coords)
+
+            target_x, target_y = int(pixel_coords[0]), int(pixel_coords[1])
+
+            # Ensure multi-channel compatibility
+            raw_multi = numpy.asarray([d.stream.raw[0] for d in streams_projections])
+            shape_y, shape_x = raw_multi.shape[-2], raw_multi.shape[-1]
+
+            # Get boundary-safe slice & crop
+            roi = get_bounding_box_slice(target_x, target_y, COM_ROI_PADDING, COM_ROI_PADDING, shape_x, shape_y)
+            multi_crop = raw_multi[(slice(None),) + roi]
```

Ruff (0.15.12)

[warning] Consider `(slice(None), *roi)` instead of concatenation

Replace with `(slice(None), *roi)`

(RUF005)